### DISCUSSION TITLE
TLCD number question

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1487] [Q&A]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/224318379-94e7a27a-1b6c-4740-ae63-5dc2832521b5.png)
![image](https://user-images.githubusercontent.com/107483730/224324366-25aadcf3-3a28-4aad-bbd1-f049f0703d87.png)
Q1:As you can see in the TLCD picture, can you tell me what a,b,c,d,e,f means?
Q2:Is the TLCD shown in the second picture only controlled by one TLCD? Or are there two TLCD controls?

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

Here are my responses:
Q1. The TLDC is defined by area rather than specific cross-sectional lengths.  `A = a*b` (`area_x` or `area_Y`).  `A_ratio = c*d/a*b` (`area_ratio_X` or `area_ratio_Y`).  `e` and `f` are not explicitly defined, only `L` and `B` are defined.
Q2. Yes, there are two TLCD controls--one along X and one along Y.

Best regards,

**Reply by icnosvbovwb** [reply]
Dear @jjonkman 
If I only consider TLCD in one direction, for example, I don't want control in X direction, how to cancel TLCD in X direction？

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

You can enable/disable the various TMD degrees of freedom through the `StC_X_DOF`, `StC_Y_DOF`, and `StC_Z_DOF` flags in the Structural Control (StC) input file.

Best regards,

**Reply by icnosvbovwb** [reply]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/227524590-4d4b0278-1cd4-4975-bf82-4ec930b5ea8c.png)
I just want to use TLCD to control the tower top displacement in the X direction. Is my setting correct?
But I see that by enabling/disabling the variable TMD degrees of freedom through the StC_ X_ DOF, StC_ Y_ DOF, and StC_ Z_ DOF to enable or disable TLCD, where the flag is (Used only when StC_ DOF_ MODE=1)，Does this have no effect on setting 3?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

That is correct.  The `StC_X_DOF`, `StC_Y_DOF`, and `StC_Z_DOF` flags are only used for the independent X-, Y-, or Z-direction TMDs (`StC_DOF_MODE` = 1); they are not used for the TLCDs.

I'm not too familiar with the TLCD option available in the StC module, but from what I can tell, when you enable TLCD (`StC_DOF_MODE` = 3), you'll get both a fore-aft (X) and side-to-side (Y) TLCD without the ability to disable either direction.

Best regards,

**Reply by icnosvbovwb** [reply]
Dear @jjonkman 
Assuming that I only use TLCDs in the X direction instead of using TLCDs in the Y direction, can I keep the parameters in the Y direction very small, so that there is approximately no existence of TLCDs in the Y direction，right?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Again, I'm not an expert on TLCDs, but I would guess you want to set the Y-direction TLCD to have negligible mass and negligible motion by setting `rho_Y` to near zero and `headLossCoeff_Y` to a very high value.

Best regards,

**Reply by icnosvbovwb** [reply]
Dear @jjonkman 
![image](https://github.com/OpenFAST/openfast/assets/107483730/c0764ede-10ca-40a2-b500-870b7fadd683)
![image](https://github.com/OpenFAST/openfast/assets/107483730/06e80810-7b06-4981-a75a-e6d61870ec11)
I applied TLCD on the top of the tower, and set rho_Y to near zero and headLossCoeff_Y to a very high value. From the results, there is still a control force in the Y direction. Can you tell me where this control force comes from?

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/225309100-2fd367e4-adc5-4518-bcb6-09c9226118ce.png)
![7a94ffcfa67f3738bc22c836044075b](https://user-images.githubusercontent.com/107483730/225308742-100226e1-463d-4f42-81ca-fd3bce3c7f1d.jpg)
af10f2278.png)
As shown in the figure, can I determine one of the baffle opening sizes according to the table? Can I choose any one(A0/A) , right? Because I can't find the control parameter setting position of this baffle in the STC control file

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

The orifice is characterized by the head-loss coefficient inputs (`headLossCoeff_X`, `headLossCoeff_Y`) in the StC input file.

Best regards,

**Comment by icnosvbovwb** [comment]
Deae @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/227960889-6d978521-ea24-4e32-9fe9-bfeed1bfdc72.png)
![image](https://user-images.githubusercontent.com/107483730/227964471-ee425d04-bf59-4471-af7c-13d06d4c88ae.png)
![dcb0ddc4ae71804611cc99ebd9abe02](https://user-images.githubusercontent.com/107483730/227965038-50674f3f-710e-4909-9c92-2f0c2f48eae6.jpg)
I have roughly calculated the various parameters of the TLCD based on the tower top frequency of OC4, and placed the TLCD at a position of 77.6m, but the effect is not obvious. Can you provide some suggestions?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Again, I'm not an expert on TLCDs and can't comment on their design.  Just a couple comments:

- To gain confidence in your use of the model, I would suggest trying to regenerate previously published results that used this model, such as those by Semyung Park of the University of Massachusetts-Amherst.
- I know the TMD model has been more widely used than the TLCC model, so, perhaps it is easiest of you design a nacelle-based damper with a TMD first, and then mimic that response with a TLCD-based damper.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://github.com/OpenFAST/openfast/assets/107483730/0112f7f3-aa8e-4a98-a18c-60336725cdd7)
![image](https://github.com/OpenFAST/openfast/assets/107483730/b6463aa5-9c59-4013-8e06-761055b2dea4)
May I ask if TStC1_Fxi and TStC1_Fyi represents the control force in the X and Y directions of a TLCD? Or do they represent the control force of each of the two TLCDs? Is pitcure 1 correct or pitcure 2 correct?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

The Structural Control (StC) outputs you refer to are the total forces in the X and Y coordinates in the inertial frame applied to the tower from the first tower-base StC; these forces include the contribution from both the X and Y TLCDs.  So, neither of your pictures is correct.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
Thanks for your reply. I have another small question：can Fast.farm be run using Simulink?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

NREL has not yet developed an interface between FAST.Farm and Simulink.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://github.com/OpenFAST/openfast/assets/107483730/c0ef8e10-c7f3-45c3-8778-c5fa8a5b4d59)
![image](https://github.com/OpenFAST/openfast/assets/107483730/4221a4bd-ad47-447e-81b2-e5c32ae6b06b)

Similarly, I set TLCD on the top of the tower and TMD in the Y direction on the platform, but I don't know where the force in the X direction on the platform comes from? The displacement of TMD in the X-direction of the platform is 0.

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

A TMD that moves only in the y direction can still cause a force in the x direction, that is, the force preventing the mass from moving in the x direction in the presence of other motions of the tower top.  Presumably you tower top has some motion that results in TMD forces in the x direction.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://github.com/OpenFAST/openfast/assets/107483730/c0764ede-10ca-40a2-b500-870b7fadd683)
![image](https://github.com/OpenFAST/openfast/assets/107483730/06e80810-7b06-4981-a75a-e6d61870ec11)
  I only applied TLCD  in the X direction on the top of the tower, and set rho_Y to near zero and headLossCoeff_Y to a very high value. From the results, there is still a control force in the Y direction. Similarly, The force preventing the TLCD from moving in the Y direction in the presence of other motions of the tower top. Right？

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Yes, I agree with your interpretation.

Best regards,
